jerry-port/default/defaultx-handler.c

```diff
@@ -25,4 +33,14 @@ void
 jerryx_port_handler_print_char (char c) /**< the character to print */
 {
   printf ("%c", c);
+
+#ifdef JERRY_DEBUGGER
+  debug_buffer[debug_buffer_index++] = c;
+
+  if ((debug_buffer_index == DEBUG_BUFFER_SIZE) || (c == '\n'))
```

Comment:
What would happen if there is only one character to print out?

Reply:
Print is always terminated by a newline. This function is the port implementation helper of the print handler extension. We know how that handler works.

Reply:
Ah... that is true.